noxfile.py

```diff
@@ -81,6 +81,13 @@ def integration_tests(session):
     session.run("pytest", "-x", "-m", "integration", "--durations=10", *session.posargs)
 
 
+@nox.session(python=python_versions)
+def self_test(session):
+    install_groups(session)
+    venv_path = session.run("poetry", "env", "info", "--path", silent=True).strip()
```

Comment:
This looks good, but since Nox usually manages a separate virtualenv per session (that's what the `install_groups(session)` sets up), I'd prefer if the `--pyenv` we pass to FawltyDeps is the one that _Nox_ has set up [^1], and not the bigger Poetry dev env. We can get this via `session.virtualenv.location`.

That said, this only applies when we run Nox locally. In GitHub CI we run `nox` with `--no-venv`[^2], and when that is the case `session.virtualenv` no longer has a `.location` member. In this case, Nox will simply run the commands (`fawltydeps` in our case) in the _same_ virtualenv that Nox itself is running in, so we can fall back to `sys.prefix` (which will point at the surrounding Poetry environment).

In short:
```suggestion
    # For --pyenv, use Nox's virtualenv, or fall back to current virtualenv
    venv_path = getattr(session.virtualenv, "location", sys.prefix)
```

[^1]: For one, the virtualenv created by Nox usually only contains the minimal dependencies (as opposed to the entire dev shell managed by Poetry), and it's good to check that this minimal env still covers what we need for FawltyDeps to run cleanly.

[^2]: The reason for using `--no-venv` in GitHub CI is somewhat complicated, related to GitHub caching, and explained in more detail in 2a57243ebb8a30640292652fe91e679723c8ef6f.